common/test/acceptance/tests/lms/test_oauth2.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for OAuth2 permission delegation.
+"""
+from ...pages.lms.oauth2_confirmation import OAuth2Confirmation
+from ...pages.lms.auto_auth import AutoAuthPage
+from bok_choy.web_app_test import WebAppTest
+
+from urlparse import urlparse, parse_qsl
+
+
+class OAuth2PermissionDelegationTests(WebAppTest):
+    """
+    Tests for acceptance/denial of permission delegation requests.
+    """
+
+    def setUp(self):
+        super(OAuth2PermissionDelegationTests, self).setUp()
+        self.oauth_page = OAuth2Confirmation(self.browser)
+
+    def _auth(self):
+        """
+        Authenticate the user.
+        """
+        AutoAuthPage(self.browser).visit()
+
+    def _qs(self, url):
+        """
+        Parse url's querystring into a dict.
+        """
+        return dict(parse_qsl(urlparse(url).query))
+
+    def test_error_for_invalid_scopes(self):
+        """
+        Requests for invalid scopes throw errors.
+        """
+        self._auth()
+        self.oauth_page.scopes = ('email', 'does-not-exist')
+        assert self.oauth_page.visit().is_browser_on_page()
+
+        self.assertTrue(self.oauth_page.has_error)
+        self.assertIn('not a valid scope', self.oauth_page.error_message)
+
+    def test_cancelling_redirects(self):
+        """
+        If you cancel the request, you're redirected to the redirect_url with a
+        denied query param.
+        """
+        self._auth()
+        assert self.oauth_page.visit().is_browser_on_page()
+        example_org = self.oauth_page.cancel()
+        example_org.wait_for_page()
+
+        query = self._qs(self.browser.current_url)
+        self.assertEqual('access_denied', query['error'])
+
+    def test_accepting_redirects(self):
+        """
+        If you accept the request, you're redirected to the redirect_url with
+        the correct query params.
+        """
+        self._auth()
+        assert self.oauth_page.visit().is_browser_on_page()
+        example_org = self.oauth_page.confirm()
```

Comment:
Done. We're now redirecting to http://does-not-exist/ instead, which will fail.. but we don't particularly care. We only want the resulting query params on the URI.